### DISCUSSION TITLE
include search parameter within action=stats

### DISCUSSION THREAD

**Discussion by adnaanqureshi** [YOURLS/YOURLS#3212] [Q&A]
Hi . 
Thanks for all the hard work providing and maintaining this great software.
I have a question about stats. I have tried using the action=stats endpoint but it returns pretty blunt data - just the total numbers. 
Is there any way of adding more search options to this endpoint? for example, id like to return the number of short links when the longurl contains the term "xxx". This is because i need to split and analyse my stats by a specific query string parameter within the long urls.
Ive looked at the API Action plugin which gives an example of creating a delete entry endpoint, but i have no idea which functions are available to allow me to create a new search parameter in the api.
Would appreciate any help or pointing in the right direction

**Comment by ozh** [comment]
Any function, PHP or YOURLS, is available.

* https://github.com/YOURLS/awesome-yourls : several plugins expand the API
* https://github.com/YOURLS/YOURLS/wiki/How-to-write-SQL-queries-in-YOURLS
* http://yourls.org/xref/_functions/index.html
